internal/report/report.go

```diff
@@ -119,6 +125,11 @@ func (r *Reporter) SendReport(ctx context.Context) (*DataResponse, error) {
 	}
 }
 
+// SetAdditionalData sets additional information to be included in the telemetry report
+func (r *Reporter) SetAdditionalData(additionalData map[string]string) {
```

Comment:
I think maybe it would be better to go with separate, per-field additions? Like,
```go
func (r *Reporter) SetAdditionalData(key string, add any) {
```
So that the _second_ added piece of data, if there ever is some, doesn't need to know the _first_ one. If we're passing a map here, it's all or nothing.

What do you think?

Reply:
Also, if we registered functions, we could nicely decouple the report data gathering from computing that data. Like:
```go
m.reporter.SetAdditionalData("min_compatible_version", func() (any, error) {
	caps := ast.CapabilitiesForThisVersion()
	minimumCompatibleVersion, _ := caps.MinimumCompatibleVersion()
	return minimumCompatibleVersion, nil
})
```
In the reporting loop, we'd invoke all registered functions and add their data to the payload.

We should probably consider adding a context.Context argument there, so we can limit the time it takes for any registered function to come up with some data (by passing in a ctx with a deadline).

Reply:
Why not keep the map and just fold it in, like

```go
for k, v := range additionalData {
    r.additionalData[k] = v
}
```

This seems like it would be the best of both worlds? 

It's too bad you can do variadics with some kind of tuple... `SetAdditionalData(data ...(string, any))` would be nice. I poked around a bit and couldn't think of a particularly clean way to handle that in Go.

Reply:
You can do `[2]any...` but it's not great 🙃

Reply:
I've [done](https://git.sr.ht/~charles/rq/tree/07870bc8c9fd1bd2e0fee79283134801f7d11f5f/item/util/util.go#L127) that sort of thing, but then you have to decide what "convert this `any` to a string" means. I guess there's always `fmt.Sprintf("%+v, val)` :vomiting_face: .

Reply:
(I guess it's worth mentioning for posterity that you can also just marshal to JSON, but then you end up with extra double quotes in your strings)

Reply:
Ok I've updated the reporter so that callers can register functions to add in extra data. Thanks for the suggestion @srenatus!